File: Homeworks/01.CSharp-Basic/06.Loops/16.DecimalToHexadecimal/DecimalToHexadecimal.cs
using System;

/* Decimal to Hexadecimal Number
 *  Using loops write a program that converts an integer number to its hexadecimal representation.
 *  The input is entered as long. The output should be a variable of type string.
 *  Do not use the built-in .NET functionality.
 */

class DecimalToHexadecimal
{
    /// GetChar(15) -> 'F'
    static char GetChar(long i)
    {
        if (i >= 10) return (char)('A' + i - 10);
        else return (char)('0' + i);
    }

    static string Base10ToBase16(long d)
    {
        string h = String.Empty;

        for (; d != 0; d /= 16) h = GetChar(d % 16) + h;

        return h;
    }

    static void Main()
    {
        Console.WriteLine(Base10ToBase16(338583669684));
    }
}

File: Homeworks/02.Java-Basic/02.Java-Syntax/src/DecimalToHexadecimal.java
import java.util.Scanner;

// 5. Write a program that enters a positive integer number num and converts and prints it in hexadecimal form.
//    You may use some built-in method from the standard Java libraries. 

public class DecimalToHexadecimal {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		
		System.out.print("Enter a number: ");
		String input = in.next();
		
		in.close();
		
		int decimal = Integer.parseInt(input);
		String hexadecimal = Integer.toHexString(decimal);
		
		System.out.println("Hex value: " + hexadecimal.toUpperCase());
	}
}